src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -383,7 +398,12 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         let defaultPythonKernel: VSCodeNotebookController | undefined;
         let defaultPythonLanguageKernel: VSCodeNotebookController | undefined;
         controllers.forEach((item) => {
-            if (item.connection.kind !== 'startUsingRemoteKernelSpec') {
+            // Sort out interactive or non-interactive controllers
+            if (
```

Comment:
Another existing bug I ran into fixing tests. Remote wasn't checking the view type. So it was possibly returning interactive controllers for notebooks and visa versa.